carapace-server/src/main/java/org/carapaceproxy/server/backends/BackendHealthManager.java

```diff
@@ -50,11 +52,12 @@
     private final StatsLogger mainLogger;
 
     private ScheduledExecutorService timer;
-    // configured only at start
-    private int period;
+    private ScheduledFuture<?> scheduledFuture;
 
     // can change at runtime
-    private volatile int connectTimeout = 60000;
+    private volatile AtomicInteger period = new AtomicInteger(0);
```

Comment:
volatile and AtomicInteger does not make sense together.

We don't need AtomicInteger, we aren't performin compare and set operations and every access to period is inside a synchronized block